smt/applications/podi.py

```diff
@@ -0,0 +1,475 @@
+
+        svd = PCA(svd_solver="randomized", random_state=random_state)
+
+        if n_modes is not None:
+            self.n_modes = n_modes
+            choice_svd = "mode"
+
+        if tol is not None:
+            if choice_svd is not None:
+                raise ValueError(
+                    "pod can't use both arguments 'n_modes' and 'tol' at the same time"
+                )
+            else:
+                choice_svd = "tol"
+
+        if choice_svd is None:
+            raise ValueError(
+                "either one of the arguments 'n_modes' and 'tol' must be specified"
+            )
+
+        database = ensure_2d_array(database, "database")
+
+        self.n_snapshot = database.shape[0]
+        self.ny = database.shape[1]
+
+        svd.fit(database)
+        self.left_basis = svd.components_
+        self.singular_values = svd.singular_values_
+        EV_list = svd.explained_variance_
+
+        if choice_svd == "tol":
+            self.n_modes = PODI.choice_n_modes_tol(EV_list, tol)
+        else:
+            if self.n_modes > self.n_snapshot:
+                raise ValueError(
+                    "the number of kept modes can't be superior to the number of data values (snapshots)"
+                )
+        self.EV_ratio = sum(EV_list[: self.n_modes]) / sum(EV_list)
+
+        self.mean = np.atleast_2d(database.mean(axis=0))
+        self.basis = self.left_basis[: self.n_modes]
+        self.coeff = np.dot(database - self.mean, self.basis.T)
+
+        self.pod_computed = True
+        self.interp_options_set = False
+        self.training_values_set = False
+
+    def get_left_basis(self) -> np.ndarray:
+        """
+        Getter for the left basis of the POD.
+
+        Returns
+        -------
+        left_basis : np.ndarray
+            Left basis of the POD.
+        """
+        return self.left_basis
+
+    def get_singular_values(self) -> np.ndarray:
+        """
+        Getter for the singular values from the Sigma matrix of the POD.
+
+        Returns
+        -------
+        singular_values : np.ndarray
+            Singular values of the POD.
+        """
+        return self.singular_values
+
+    def get_ev_ratio(self) -> float:
+        """
+        Getter for the explained variance ratio with the kept modes.
+
+        Returns
+        -------
+        EV_ratio : float
+            Explained variance ratio with the current kept modes.
+        """
+        return self.EV_ratio
+
+    def get_n_modes(self) -> int:
+        """
+        Getter for the number of modes kept during the POD.
+
+        Returns
+        -------
+        n_modes : int
+            number of modes kept during the POD.
+        """
+        return self.n_modes
+
+    def get_svd(self):
+        """
+        Getter for the svd object.
+
+        Returns
+        -------
+        svd
+        """
+        return self.svd
+
+    def set_interp_options(
+        self, interp_type: str = "KRG", interp_options_list: list = [{}]
```

Comment:
As you type the arguments, I think you can shorten the name and keep just `interp_options`